### ISSUE TITLE
[LINT_BUG]: box_default_linters$box_usage_linter errors when it shouldn't

### ISSUE THREAD

**Issue by SaintRod** [Appsilon/box.linters#146]
### box.linters version

0.10.5

### Sample source code to lint

```
box::use(
    here[here]
)

# fails
for (file in here()) {
  assign(
    x = paste0("path_", tolower(basename(file))),
    value = file,
    envir = .GlobalEnv
  )
  rm(file)
}

# succeeds
tmpFunc = function(name) paste0("path_", tolower(basename(name)))
for (file in here()) {
  assign(
    x = tmpFunc(file),
    value = file,
    envir = .GlobalEnv
  )
  rm(file)
}
```

### Lint command used

```
box::use(
  box.linters[box_default_linters],
  here[here],
  lintr[lint],
)

lint(
  filename = here("test.r"),
  linters = box_default_linters$box_usage_linter
)
```

### Lint result

```
Error: Linter 'box_usage_linter' failed in ./test.r: <text>:4:1: unexpected ','
3: "path_"
4: ,
   ^
```

### Expected result

No error expected

**Comment by SaintRod** [comment]
PS. I'm not using box.linters with rhino

**Comment by radbasa** [comment]
```R
assign(
  x = paste0("path_", tolower(basename(file))),
  value = file,
  envir = .GlobalEnv
)
```

This fails linting because the linter expects the value of `x` to be a character string. Because linting does static code analysis, it cannot and does not process or execute the `paste()` command to build the character string.

The other way you presented won't trigger a linting error, but it may cause linting errors further down your code. `box_usage_linter()` may complain about:

```R
x <- path_some_file
```

As `path_some_file` was not defined previously. A different way to accomplish this might be to do the following:

```R
file_paths <- list()
assign(
  x = file_paths[[tolower(basename(file))]],
  value = file,
  envir = .GlobalEnv
)

# This should be ok
foo <- file_paths$some_filename
```

**Comment by SaintRod** [comment]
Thanks @radbasa 

> This fails linting because the linter expects the value of x to be a character string. Because linting does static code analysis, it cannot and does not process or execute the paste() command to build the character string.

Is it possible for the linter to know that the output of paste and paste0 is of class "character"? This way no code needs to be executed and the type requirement for the x arg in `assign()` is satisfied.

**Comment by radbasa** [comment]
This error is being thrown in`lintr::get_r_string()`

https://github.com/r-lib/lintr/blob/a2b8d6fec967e037e736ba81eb7fca83333438c7/R/utils.R#L226-L243
